### ISSUE TITLE
Loudness Normalization Option Not Working

### ISSUE THREAD

**Issue by GKid94** [alextrv/enhanced-h264ify#1]
I still have the loudness normalization when I disable the option

![60550594-c91ec380-9d28-11e9-87db-c777829f0076](https://user-images.githubusercontent.com/18757054/60943700-09f97800-a2e7-11e9-9ec8-ee4f5538ef9d.jpg)

**Comment by alextrv** [comment]
The text about normalization in "Stats for nerds" isn't supposed to be changed. The extension changes volume internally. So it doesn't matter what is shown in "Stats for nerds".

**Comment by GKid94** [comment]
Yes it should, it should only say 100/100 volume without the brackets like this

![m0FAUdb](https://user-images.githubusercontent.com/18757054/61100853-ec9fe780-a467-11e9-93de-84e52fc76292.jpg)

**Comment by alextrv** [comment]
The information inside parentheses is just a cosmetic thing in this case. If you worry only about that, you shouldn't. As I wrote before the extension changes volume level internally by setting it directly in video tag. If I tried to get rid of this text, the extension would depend on YouTube implementation more than it does now. So if YouTube developers made some changes in their code, there was a higher chance that disabling loudness normalization might not work. It's just more reliable solution.

**Comment by GKid94** [comment]
So the extension keeps every video at 100/100 volume instead of 100/45 for example but it says in the brackets that the volume was lowered but it's not

**Comment by GKid94** [comment]
I have the option enabled and it says that the volume is normalized from 9 to 5, and it sounds the same as without the extension

![4535345](https://user-images.githubusercontent.com/18757054/61166756-50dbad80-a533-11e9-86f8-0abf4d784026.jpg)

**Comment by maxdrash** [comment]
For me, I can hear a normalization difference with or without the checkbox.

**Comment by GKid94** [comment]
Because the option to disable the loudness normalization is not working, you can double right click on the video and select "show controls" and when you hover the cursor over the sound icon you can see if the video volume is lowered or not

**Comment by GKid94** [comment]
@alextrv @maxdrash

**Comment by alextrv** [comment]
> Because the option to disable the loudness normalization is not working, you can double right click on the video and select "show controls" and when you hover the cursor over the sound icon you can see if the video volume is lowered or not

It does work. I've made a video to show you.

https://tinyurl.com/y43qjhs4

**Comment by GKid94** [comment]
@alextrv Try it on more videos

**Comment by Onurtag** [comment]
It definitely works for me.  
You can see the actual volume of the video in the console. With the option off it will be lower on normalized videos.  
`document.querySelector('video').volume`

**Comment by GKid94** [comment]
@Onurtag Try it on more videos

**Comment by Onurtag** [comment]
@GKid94 give your non-working example video?

**Comment by GKid94** [comment]
https://youtu.be/umD_6EhHlek

![3425345](https://user-images.githubusercontent.com/18757054/63647170-760c3000-c71d-11e9-9e74-63f141e45ac3.jpg)

**Comment by GKid94** [comment]
@Onurtag @alextrv @maxdrash  https://youtu.be/umD_6EhHlek

![3425345](https://user-images.githubusercontent.com/18757054/63647180-a48a0b00-c71d-11e9-85cb-1a48588e8791.jpg)

**Comment by Onurtag** [comment]
I think you are partly right.  
It doesn't work if the page takes too long to load.  
The correct volume might have to be continuously applied [like this](https://webapps.stackexchange.com/questions/90995/how-can-i-change-a-youtube-videos-volume-beyond-the-normal-limits/103665#103665) but I don't know how taxing requestanimationframe is or if there is a better way to do it.  

I was using an userscript from the above example before I found this extension.  https://gist.github.com/Onurtag/28d8826e2f7e369a3fe57bd69bb61f8f

**Comment by alextrv** [comment]
Okay, I found a bug.
Steps to reproduce:
1. Activate the option "Disable Loudness Normalization"
2. Open a new tab and go to youtube.com
3. Search any video and click on it
4. Double right click on video, select "Show controls" and you will see that it doesn't work.

But if you reload the tab it should work. Probably that's because youtube uses ajax without reloading pages and it breaks the extension. I will try to fix it soon. So workaround for now: just reload the tab.

**Comment by GKid94** [comment]
@alextrv Can you update the extension the option doesn't work

**Comment by GKid94** [comment]
@alextrv ?

**Comment by GKid94** [comment]
The option is still not working

![Screenshot_7](https://user-images.githubusercontent.com/18757054/74127532-3528aa80-4bdb-11ea-83e4-3e1820022ea3.png)

**Comment by Onurtag** [comment]
@GKid94 Use the userscript https://gist.github.com/abec2304/2782f4fc47f9d010dfaab00f25e69c8a

**Comment by GKid94** [comment]
How do I add it

**Comment by GKid94** [comment]
?

**Comment by GKid94** [comment]
@Onurtag It doesn't work

**Comment by Onurtag** [comment]
@GKid94 
It wont show up in the UI. The audio level works correctly. 
We all switched to this instead btw  
https://gist.github.com/abec2304/2782f4fc47f9d010dfaab00f25e69c8a

**Comment by GKid94** [comment]
@Onurtag This is with that script added, the video volume is on 100%

![Screenshot_1](https://user-images.githubusercontent.com/18757054/82136908-2c1b2b00-9813-11ea-9e35-6725e166bd1a.png)

**Comment by Onurtag** [comment]
Did you read the above comment?
It will NOT show up in the UI.

**Comment by GKid94** [comment]
@Onurtag Then how will I know if it's working or not

**Comment by GKid94** [comment]
And the script also needs to be updated

**Comment by Onurtag** [comment]
Sorry, I didn't know you were deaf. The userscript only fixes the volume not the UI.
I am unfollowing the issue. Take care and goodbye.  

@alextrv Sorry for talking in here. I think you can lock this issue because we have a [good alternative](
https://gist.github.com/abec2304/2782f4fc47f9d010dfaab00f25e69c8a) now.

**Comment by GKid94** [comment]
@Onurtag I'm not deaf and I don't hear any difference with that scrip, and you don't have to be rude

**Comment by GKid94** [comment]
@Onurtag And also the video page sometimes won't load with that script

**Comment by vladaad** [comment]
@GKid94 both the userscript and the extension work and I can hear the difference. It's either just you not hearing the difference, or something weird on your side.

**Comment by GKid94** [comment]
@vladaad Yes they work a couple of months later after I added a scrip with Tampermonkey and YouTube changed the algorithm a little

**Comment by ogmkp** [comment]
This project seems abandoned.

**Comment by mara004** [comment]
Indeed the normalization in enhanced-h264ify is broken or incomplete.
On Linux Firefox there's a bug where loudness normalization changes the Firefox volume in the system mixer, so it's easy to tell. -- Unless there's a conflict with the "Youtube Auto HD + FPS", but I don't think so.

Is anyone aware of an up-to-date solution to get rid of this normalization?

**Comment by mara004** [comment]
Update: This works for me:
https://gist.github.com/fa7ad/fa995474f5cb9fe91fb209686881373d

**Comment by mara004** [comment]
Whoops, seems like https://github.com/alextrv/enhanced-h264ify/issues/7 did the trick.
